### ISSUE TITLE
todo - run tests against docker desktop on windows

### ISSUE THREAD

**Issue by alexanderankin** [testcontainers/testcontainers-python#621]
from discussion in #610

**Comment by black-snow** [comment]
Let me know if I can assist you. Ain't got too much time to spare but I'll do my best ;)

**Comment by AlexElizard** [comment]
My little research.

Introductory:
1. [Fedora Workstation 42 x86_64](https://fedoraproject.org/ru/workstation/download)
2. [Docker Desktop](https://docs.docker.com/desktop/setup/install/linux/fedora/)

Let's look at running a basic example from the [official website](https://testcontainers-python.readthedocs.io/en/latest/#getting-started)
```python
from testcontainers.postgres import PostgresContainer
import sqlalchemy

with PostgresContainer("postgres:16") as postgres:
    psql_url = postgres.get_connection_url()
    engine = sqlalchemy.create_engine(psql_url)
    with engine.begin() as connection:
        version, = connection.execute(sqlalchemy.text("SELECT version()")).fetchone()

print(version)
```

When DockerContainer is initialized, [DockerClient](https://github.com/testcontainers/testcontainers-python/blob/bb646e903236a1df72bc38dbb47d1dba95527198/core/testcontainers/core/container.py#L85) is called, which gets the `docker_host` from [tc.host in the ~/.testcontainers.properties file or from the DOCKER_HOST environment variable](https://github.com/testcontainers/testcontainers-python/blob/bb646e903236a1df72bc38dbb47d1dba95527198/core/testcontainers/core/docker_client.py#L268-L269). 

Because of this, attempting to start fails with the error `docker.errors.DockerException: Error while fetching server API version: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))`. 

This problem can be solved in `~/.bashrc` by setting the environment variable `export DOCKER_HOST=unix://${HOME}/.docker/desktop/docker.sock` or in `~/.testcontainers.properties` by setting `tc.host==unix://home/username/.docker/desktop/docker.sock`. By the way, `TESTCONTAINERS_DOCKER_SOCKET_OVERRIDE` environment does not solve the problem.

Then we start to crash with the error `docker.errors.APIError: 500 Server Error for http+docker://localhost/v1.51/containers/c57d89b2acb9f9fd4997576bdf62204daacbdfcf01006555cd0cda9e7f3624f7/start:
Internal Server Error ("mounts denied: The path /socket_mnt/home/alexelizard/.docker/desktop/docker.sock is not shared from the host and is not known to Docker.
You can configure shared paths from Docker -> Preferences... -> Resources -> File Sharing. See https://docs.docker.com/ for more info.")`

This is an error [on startup Ryuk](https://github.com/testcontainers/testcontainers-python/blob/bb646e903236a1df72bc38dbb47d1dba95527198/core/testcontainers/core/container.py#L176). I couldn't figure out where Ryuk gets the `/socket_mnt`.
Using `export TESTCONTAINERS_DOCKER_SOCKET_OVERRIDE=${HOME}/.docker/desktop/docker.sock` doesn't fix the issue with `/socket_mnt`. 

The solution was found in [testcontainers-java #7036](https://github.com/testcontainers/testcontainers-java/issues/7036#issuecomment-2289190880) - you need to set the environment variable `export TESTCONTAINERS_DOCKER_SOCKET_OVERRIDE=host.docker.internal`.

In summary: to run testcontainers via Docker Desktop in Fedora Workstation, you need to set the following environment variables in `~/.bashrc`:
```bash
export DOCKER_HOST=unix://${HOME}/.docker/desktop/docker.sock
export TESTCONTAINERS_DOCKER_SOCKET_OVERRIDE=host.docker.internal
```

Hope this helps.

**Comment by kiview** [comment]
Btw., it's totally ok to explicitly document, that Docker Desktop for Linux is not officially supported by tc-python (we have a similar situation for the other languages, also since having DD4L on CI is very _hard_).

**Comment by AlexElizard** [comment]
I'm not sure this is correct. Docker Desktop is quite widely used. Moreover, according to the [documentation](https://github.com/testcontainers/testcontainers-python/blob/bb646e903236a1df72bc38dbb47d1dba95527198/docs/features/configuration.md#docker-host-detection) the Docker host should be detected automatically (item 6), but this doesn't happen.
